utility/rp_client.py

```diff
@@ -61,23 +63,65 @@
 
 log = logging.getLogger(__name__)
 doc = """
-Standard script to push all the logs from Xunit files to Report Portal
+Standard script to push all the logs from Xunit files and get launcher details from Report Portal
 
     Usage:
         rp_client.py --config_file <str> --payload_dir <str>
+        rp_client.py --config_file <str> --launch_id <int>
+        rp_client.py --config_file <str> --launch_id <int> --output <file-path>
         rp_client.py (-h | --help)
 
     Options:
         -h --help          Shows the command usage
-        -c --config_file <str>         Config file with report portal details.
-        -d --payload_dir <str>   Paylod directory where we have results and attachments
+        -c --config_file <str>  Config file with report portal details.
+        -d --payload_dir <str>  Paylod directory where we have results and attachments
+        -l --launch_id <int>    Report portal launch id
+        -o --output <str>       File path to json output
 """
 
 
-def upload_logs(args):
+def get_launch_details(config_file, launch_id):
+    """
+    Get launcher details for specific launch id
+    """
+    args = { "config_file": config_file }
+    rp_host_url = os.environ.get("RP_HOST_URL", None)
+    url = "{}/ui/#cephci/launches/all/{}/{}"
+    try:
+        rportal = ReportPortalV1(PreProcClient((args)).configs.rp_config)
+        api = "item/v2?filter.level.path=1&page.page=1&page.size=50&page.sort=startTime%2CASC&providerType=launch&launchId={}"
+        results = rportal.api_get(api.format(launch_id))
+        results = json.loads(results.content).get("content")
+
+        testsuites = list()
+        for ts in results:
+            testsuite = {
+                "id": ts.get("id"),
+                "name": ts.get("name"),
+                "status": ts.get('status'),
+                "total": ts.get('statistics').get('executions').get('total'),
+                "passed": ts.get('statistics').get('executions').get('passed'),
+            }
+            testsuite.update({"failed": int(testsuite.get("total")) - int(testsuite.get("passed"))})
```

Comment:
There could be some skipped tests as well right, when we have enabled abort-on-fail, the tests after this test will not get executed at all. Is there any way we could accomodate them?

Reply:
@manasagowri if there are no failures or skipped tests then rp response don't include this value. Removed step to calculate failed tcs, as rp response returns the same in case there are any. Using default value `0` incase `failed` and `skipped` values are missing from response.